fractal_server/app/routes/api/v2/workflow_import.py

```diff
@@ -0,0 +1,230 @@
+            detail="Expected one TaskV2 to match the imported one.",
+        )
+
+    return task.id
+
+
+async def _get_task_by_source(wf_task, user, db) -> Optional[TaskV2]:
+    """Find a task by its source."""
+    if wf_task.task.source is None:
+        return None
+
+    stm_source = (
+        select(TaskV2)
+        .join(TaskGroupV2)
+        .where(TaskGroupV2.id == TaskV2.taskgroupv2_id)
+        .where(TaskV2.source == wf_task.task.source)
+        .where(
+            or_(
+                TaskGroupV2.user_id == user.id,
+                TaskGroupV2.user_group_id.in_(
+                    select(LinkUserGroup.group_id).where(
+                        LinkUserGroup.user_id == user.id
+                    )
+                ),
+            )
+        )
+    )
+    res = await db.execute(stm_source)
+    task = res.scalars().one_or_none()
+    return task
+
+
+async def _get_task_by_version(
+    wf_task, user, db, task_user_list, default_group_id
+) -> Optional[TaskV2]:
+    """Find a task by version."""
+    if wf_task.task.version is None:
+        return await _get_latest_user_task(task_user_list, db, wf_task)
+
+    stm = (
+        select(TaskV2)
+        .join(TaskGroupV2)
+        .where(TaskGroupV2.id == TaskV2.taskgroupv2_id)
+        .where(TaskGroupV2.pkg_name == wf_task.task.pkg_name)
+        .where(TaskV2.name == wf_task.task.name)
+        .where(TaskGroupV2.version == wf_task.task.version)
+        .where(
+            or_(
+                TaskGroupV2.user_id == user.id,
+                TaskGroupV2.user_group_id.in_(
+                    select(LinkUserGroup.group_id).where(
+                        LinkUserGroup.user_id == user.id
+                    )
+                ),
+            )
+        )
+    )
+    res = await db.execute(stm)
+    task = res.scalars().all()
+    if len(task) > 1:
+        return await _resolve_task_conflict(task, user, db, default_group_id)
+    elif len(task) == 1:
+        return task[0]
+    else:
+        return None
+
+
+async def _get_latest_user_task(task_user_list, db, wf_task) -> TaskV2:
+    """Retrieve the latest task for the user."""
+    if not task_user_list:
+        raise HTTPException(
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            detail="Expected one TaskV2 to match the imported one.",
+        )
+    latest_task = max(task_user_list, key=lambda t: t.version)
+
+    stm = (
+        select(TaskV2)
+        .join(TaskGroupV2)
+        .where(TaskGroupV2.id == TaskV2.taskgroupv2_id)
+        .where(TaskGroupV2.pkg_name == wf_task.task.pkg_name)
+        .where(TaskV2.name == wf_task.task.name)
+        .where(TaskGroupV2.version == latest_task.version)
+    )
+    res = await db.execute(stm)
+    task = res.scalars().one_or_none()
+    return task
+
+
+async def _resolve_task_conflict(
+    task_list, user, db, default_group_id
+) -> TaskV2:
+    """Resolve conflicts when multiple tasks match."""
+    for task in task_list:
```

Comment:
Priorities:
1. A task with user_id matching to mine
2. "All" group
3. user groups by age

Reply:
[
{user_group_id: 1},
{user_group_id: 2},
]

query LinkUserGroup with group_id in {1, 2} and user_id=mine - sorted by timestamp